src/hotspot/share/gc/shenandoah/shenandoahCodeRoots.cpp

```diff
@@ -126,17 +126,7 @@ void ShenandoahCodeRoots::flush_nmethod(nmethod* nm) {
 }
 
 void ShenandoahCodeRoots::arm_nmethods() {
-  assert(SafepointSynchronize::is_at_safepoint(), "Must be at a safepoint");
-  _disarmed_value ++;
-  // 0 is reserved for new nmethod
-  if (_disarmed_value == 0) {
-    _disarmed_value = 1;
-  }
-
-  JavaThreadIteratorWithHandle jtiwh;
-  for (JavaThread *thr = jtiwh.next(); thr != NULL; thr = jtiwh.next()) {
-    ShenandoahThreadLocalData::set_disarmed_value(thr, _disarmed_value);
-  }
+  BarrierSet::barrier_set()->barrier_set_nmethod()->arm_all_nmethods();
```

Comment:
Question: in `ShenandoahBarrierSet::on_thread_attach`, we check `barrier_set_nmethod() != NULL`, do we need to do the same check here, or?

Reply:
Only in `passive` mode, Shenandoah does not install nmethod barrier, but `passive` mode never calls `arm_nmethod()`, etc. nmethod entry barrier related methods.